include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,421 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, TileBasedRenderer);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, UMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, CacheCoherentUMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, IsolatedMMU);
+
+    // 18: D3D12 Options2
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions2, DepthBoundsTestSupported);
+    FEATURE_SUPPORT_GET(D3D12_PROGRAMMABLE_SAMPLE_POSITIONS_TIER, m_dOptions2, ProgrammableSamplePositionsTier);
+
+    // 19: Shader Cache
+    FEATURE_SUPPORT_GET_NAME(D3D12_SHADER_CACHE_SUPPORT_FLAGS, m_dShaderCache, SupportFlags, ShaderCacheSupportFlags);
+
+private: // Private helpers
+    
+    // Helper function to decide the highest shader model supported by the system
+    // Stores the result in m_dShaderModel
+    // Must be updated whenever a new shader model is added to the d3d12.h header
+    HRESULT QueryHighestShaderModel()
+    {
+        // Check support in descending order
+        HRESULT result;
+
+        // SHADER_MODEL_6_X
+        for (int i = D3D_SHADER_MODEL_6_7; i >= D3D_SHADER_MODEL_6_0; i--) {
+            m_dShaderModel.HighestShaderModel = static_cast<D3D_SHADER_MODEL>(i);
+            result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &m_dShaderModel, sizeof(D3D12_FEATURE_DATA_SHADER_MODEL));
+            if (SUCCEEDED(result) || result != E_INVALIDARG) {
+                // Indicates that the version is recognizable by the runtime. 
+                // The highest version supported has been written in the member data structure
+                // Also terminate on unexpected error code
+                return result;
+            }
+        }
+
+        // SHADER_MODEL_5_1
+        // This is the last model included in the d3d12.h header.
+        m_dShaderModel.HighestShaderModel = D3D_SHADER_MODEL_5_1;
+        result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &m_dShaderModel, sizeof(D3D12_FEATURE_DATA_SHADER_MODEL));
+        if (result == E_INVALIDARG) {
+            // Shader model may not be supported
+            m_dShaderModel.HighestShaderModel = (D3D_SHADER_MODEL)0;
+            result = S_OK; // Continue initialization
+        }
+
+        return result;
+    }
+
+    // Helper function to decide the highest root signature supported
+    // Must be updated whenever a new root signature version is added to the d3d12.h header
+    HRESULT QueryHighestRootSignatureVersion()
+    {
+        HRESULT result;
+
+        // ROOT_SIGNATURE_VERSION_1_X
+        for (int i = D3D_ROOT_SIGNATURE_VERSION_1_1; i >= D3D_ROOT_SIGNATURE_VERSION_1; i--) {
+            m_dRootSignature.HighestVersion = static_cast<D3D_ROOT_SIGNATURE_VERSION>(i);
+            result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_ROOT_SIGNATURE, &m_dRootSignature, sizeof(D3D12_FEATURE_DATA_ROOT_SIGNATURE));
+            if (SUCCEEDED(result) || result != E_INVALIDARG) {

```

Comment:
here too, can simplify to result != E_INVALIDARG

Reply:
Resolved